CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -2290,8 +2290,20 @@ + (void)showTopicsDialog:(UIWindow *)targetWindow {
             }
             
             dispatch_async(dispatch_get_main_queue(), ^{
-                CPTopicsViewController *topicsController = [[CPTopicsViewController alloc] initWithAvailableTopics:channelTopics selectedTopics:[self getSubscriptionTopics] hasSubscriptionTopics:[self hasSubscriptionTopics]];
+                NSArray* topics = [self getSubscriptionTopics];
+                if (!topics || [topics count] == 0) {
```

Comment:
this looks way too nested. use early exists here please.

Reply:
also im pretty sure you can refactor this (by exporting separate code blocks into functions) so its readability increases.

Reply:
but as it is right now it's unreadable imo

Reply:
Ok I will do the needful changes and update you by EOD.